contribute/testing.md

```diff
@@ -0,0 +1,230 @@
+# Beam Testing Documentation
+
+# Overview
+
+Apache Beam (incubating) is a rapidly-maturing software project with a strong commitment to testing. Consequently, it has many testing-related needs. It requires precommit tests to ensure code going into the repository meets a certain quality bar and it requires ongoing postcommit tests to make sure that more subtle changes which escape precommit are nonetheless caught. This document outlines how to write tests, which tests are appropriate where, and when tests are run, with some additional information about the testing systems at the bottom.
+
+## Beam Java Testing Matrix
+
+
+<table>
```

Comment:
Found a decent html -> markdown converter online, but the markdown syntax table is way worse because of the links - each line winds up being hundreds of characters long. I think this is way more readable.